discretisedfield/field_new.py

```diff
@@ -77,8 +77,59 @@ def __init__(
     #    self._subregions = subregions
 
     @classmethod
-    def from_xarray(cls):  # -> still required (?)
-        raise NotImplementedError()
+    def from_xarray(cls, xa: xr.DataArray):
+
+        if not isinstance(xa, xr.DataArray):
+            raise TypeError("Argument must be a xr.DataArray.")
+
+        if len(xa.coords) != xa.data.ndim:
+            raise AttributeError(
+                "The xr.DataArray must have the same number of coordinates as"
+                " the axes of underlying data array."
+            )
+
+        for i in xa.coords.dims[:-1]:
+            if xa[i].data.size > 1 and not np.allclose(
+                np.diff(xa[i].data), np.diff(xa[i].data).mean()
+            ):
+                raise ValueError(f"Coordinates of {i} must be equally spaced.")
+
+        try:
+            cell = xa.attrs["cell"]
+        except KeyError:
+            if any(len_ == 1 for len_ in xa.data.shape[:-1]):
+                raise KeyError(
+                    "DataArray must have a 'cell' attribute if any "
+                    "of the geometric directions has a single cell."
+                ) from None
```

Comment:
Is `from None` necessary?

Reply:
I am trying to supress errors from the `xarray.DataArray` which might confuse the user. I can remove it if you like.